extensions/amp-image-slider/0.1/test/integration/test-amp-image-slider.js

```diff
@@ -60,50 +67,119 @@ config.run('amp-image-slider', function() {
   const experiments = ['amp-image-slider'];
   const extensions = ['amp-image-slider'];
 
-  describes.integration('default type', {
+  describes.integration('amp-image-slider', {
     body: sliderBody,
     css,
     extensions,
     experiments,
   }, env => {
     let win;
     let doc;
-    let slider;
-    let sliderImpl;
-    let rect;
-    let bar_;
-    let container_;
-    let rightMask_;
-    let leftAmpImage_;
-    let rightAmpImage_;
-    let leftLabel_;
-    let rightLabel_;
+    let observer;
+    let s1; // sliderInfo of slider 1
+    let s2; // sliderInfo of slider 2
 
     beforeEach(() => {
       win = env.win;
       doc = win.document;
+      return prep();
     });
 
-    function waitForImageSlider() {
-      return poll('wait for amp-image-slider to complete', () => {
-        slider = doc.querySelector('#slider');
+    afterEach(() => {
+      // Just in case
+      cleanupObserver();
+    });
 
-        if (!slider) {
+    // Wait for certain ms
+    function timeout(ms) {
+      return new Promise(resolve => win.setTimeout(resolve, ms));
+    }
+
+    // Ensure sliders has been properly laid out
+    function areSlidersReady() {
+      // poll() is still needed here
+      // since we never know from DOM mutation
+      // if event handlers are registered yet
+      return poll('check if sliders ready', () => {
+        const slider1 = doc.getElementById('s1');
+        const slider2 = doc.getElementById('s2');
+        if (!slider1 || !slider2) {
           return false;
         }
-
-        sliderImpl = slider.implementation_;
-
-        if (!sliderImpl) {
+        if (!slider1.signals || !slider2.signals ||
+            !slider1.signals() || !slider2.signals()) {
           return false;
         }
-        // layoutCallback is called
-        // this is the same as CommonSignal.LOAD_END
-        return !!slider.signals().get('load-end');
+        // .signals() are accessible
+        // LOAD_END is emitted after layoutCallback
+        return !!slider1.signals().get('load-end') &&
```

Comment:
Do we need `poll` here? Can we just import `CommonSignals` and use `Promise.all([slider1.signals().whenSignal(CommonSignals.LOAD_END), ...])`?

Reply:
My bad, forgot to change this line